src/map/party.cpp

```diff
@@ -335,7 +335,17 @@ void CParty::RemoveMember(CBattleEntity* PEntity)
                     }
                 }
 
-                auto trustCount = static_cast<CCharEntity*>(m_PLeader)->PTrusts.size();
+                size_t trustCount = 0;
+                if (m_PLeader != nullptr)
+                {
+                    std::vector<CTrustEntity*> PTrusts = static_cast<CCharEntity*>(m_PLeader)->PTrusts;
```

Comment:
No need to make a copy of the trusts vector here, same for below.

trustCount is init'd as zero:
```
size_t trustCount = 0;
```

The size of a vector is zero when it's empty.

So you can simplify this block (and below) with something like:

```cpp
trustCount = static_cast<CCharEntity*>(m_PLeader)->PTrusts.size();
```

Reply:
Simplified both this line and 867.